explainer-beforematch.md

```diff
@@ -0,0 +1,135 @@
+indication that the user initiated a find-in-page request, or fragment or
+scroll-to-text navigation.
+
+The `beforematch` event is a signal that is a step in the direction that allows
+us to leverage information that the user-agent has to assist the page in making
+the user experience great. Specifically, with future additions to
+display-locking functionality, it will be possible to process text for find-in-page
+match in sections that are not visible. In turn, the `beforematch` event will be
+fired on hidden (a.k.a. collapsed) sections, allowing the developer to unhide
+the section. The net effect is that the user is able to use find-in-page to find
+content in a collapsed section -- something that is not currently possible.
+
+Even without display-locking features, `beforematch` is useful signal to the
+page which allows custom styling of the matched element, which is now only
+possible with approximations from scroll positions and intersection
+observations.
+
+### Use Cases
+
+#### Example 1
+
+```html
+<style>
+.section {
+  transition: background 2s;
+}
+.highlighted {
+  background: cornsilk;
+  transition: background 0s;
+}
+</style>
+
+<div class=section>Lorem ipsum dolor sit amet</div>
+<div class=section>consectetur adipiscing elit</div>
+<div class=section>Sed augue lacus</div>
+
+<script>
+function highlight(e) {
+  e.target.classList.add("highlighted");
+  requestAnimationFrame(
+    () => requestAnimationFrame(
+      () => e.target.classList.remove("highlighted")));
+}
+
+document.querySelectorAll(".section").forEach(item => {
+  item.addEventListener("beforematch", highlight);
+});
+</script>
+```
+
+In this example, a highlighted entry gains a corksilk background color, which
+fades over approximately two seconds. It's a subtle effect that grabs the user's
+attention but is not too intrusive. 
+
+We use a `beforematch` event to highlight the section. Note that the highlighted
+element is the target element of the `beforematch` event, but we can also modify
+any related style or DOM based on the target's location.
+
+The effect of this example is that the section is briefly highlighted in
+cornsilk on an active find-in-page match. This effect happens in addition to the
+user-agent highlight the found match.
+
+#### Example 2
+```html
+<style>
+.collapsed {
+  subtree-visibility: hidden-matchable;
+}
+</style>
+
+Please explore the following sections:
+<h1>Introduction</h1> <div class=collapsed>...</div>
+<h1>Thesis</h1>       <div class=collapsed>...</div>
+<h1>Conclusion</h1>   <div class=collapsed>...</div>
+
+<script>
+function expand(e) {
+  e.target.classList.remove("collapsed");
+}
+
+document.querySelectorAll(".collapsed").forEach(item => {
+  item.addEventListener("beforematch", expand);
+});
+</script>
+```
+
+In this example, most of the content of the page is hidden in collapsed section.
+It uses a display-locking property called `subtree-visibility` to hide the
```

Comment:
```suggestion
It uses the upcoming `subtree-visibility` CSS property to hide the
```

Reply:
Also link to the privacy section of the spec for text fragment navigation for more considerations that may be relevant.

Reply:
turns out there was not much there